File: Company Interview/GG/pickSmallest.java
public class pickSmallest {
	public static void main(String[] args) {
		System.out.println(pick_smallest(233614));
		System.out.println(pick_smallest(1578));
		System.out.println(pick_smallest(135));
		System.out.println(pick_smallest(10000));
	}

	public static int pick_smallest(int num) {
		int smallest_num = Integer.MAX_VALUE;
		String num_string = String.valueOf(num);
		for (int i = 0; i < num_string.length() - 1; i++) {
			char replace_int_str = num_string.charAt(i) > num_string.charAt(i + 1) ? num_string.charAt(i) : num_string.charAt(i + 1);
			int temp = (int)Integer.valueOf(num_string.substring(0, i) 
					+ replace_int_str + num_string.substring(i + 2, num_string.length()));
			if (temp < smallest_num) {
				smallest_num = temp;
			}
		}
		return smallest_num;
	}

}

File: Company Interview/GG/pickSmallest.py
class Solution(object):
	def pickSmallest(self, num):
		current_largest_num = float("inf")
		for i in xrange(len(num) - 1):
			replace_int = num[i] if num[i] > num[i + 1] else num[i + 1]
			temp = int(num[:i] + replace_int + num[i + 2:])
			if temp < current_largest_num:
				current_largest_num = temp
		return current_largest_num

if __name__ == "__main__":
	s = Solution()
	print s.pickSmallest("233614")
	print s.pickSmallest("1578")
	print s.pickSmallest("135")
	print s.pickSmallest("10000")
